pkg/client/unversioned/restclient.go

```diff
@@ -45,26 +44,24 @@ const (
 //
 // Most consumers should use client.New() to get a Kubernetes API client.
 type RESTClient struct {
-	baseURL *url.URL
-	// A string identifying the version of the API this client is expected to use.
-	groupVersion unversioned.GroupVersion
+	// Base is the root URL for all invocations of the client
+	Base *url.URL
```

Comment:
Private is great for restricting API surface of structs. Making this public means we now support people changing this. I wouldn't want to do this without a good reason.